warehouse/manager/manager.go

```diff
@@ -43,6 +43,7 @@ type ManagerI interface {
 
 type WarehouseDelete interface {
 	DropTable(tableName string) (err error)
+	DeleteBy(tableName []string, jobRunID string, sourceID string, taskRunID string) (success bool, err error)
```

Comment:
`tableNames`. I would drop success and use err!= nil or not as the basis for it.

Reply:
Okay. Will refactor this.

Reply:
Have refactored this & completed.